git_machete/git_operations.py

```diff
@@ -645,7 +646,10 @@ def __get_merge_base(self, sha1: FullCommitHash, sha2: FullCommitHash) -> FullCo
             #   then there is exactly one merge-base - the ancestor,
             # * if neither of sha1, sha2 is an ancestor of another,
             #   then none of the (possibly more than one) merge-bases is equal to either of sha1/sha2 anyway.
-            self.__merge_base_cached[sha1, sha2] = FullCommitHash.of(self._popen_git("merge-base", sha1, sha2).rstrip())
+            # In the rare case when sha1, sha2 have no common commits, the flag: allow_non_zero=True
+            # (allows, non zero exit code to be returned by git merge-base command, without raising an exception)
+            # is used and the __get_merge_base function returns None.
```

Comment:
Generally `Optional`s are still treated very liberally coz we haven't applied mypy strict mode yet (#118)... but I'd change the return type of this method to `Optional[FullCommitHash]` while we're here, even though mypy doesn't complain yet

Reply:
This method has return type `Optional[FullCommitHash]` already, no need to change it :)

Reply:
Hmmm wait
```python
def __get_merge_base(self, sha1: FullCommitHash, sha2: FullCommitHash) -> FullCommitHash:
```